resolwe/flow/tests/test_api.py

```diff
@@ -301,6 +301,35 @@ def test_change_descriptor_schema(self):
         collection.refresh_from_db()
         self.assertEqual(collection.descriptor_schema, d_schema)
 
+    def test_change_slug(self):
+        collection1 = Collection.objects.create(name="Collection", contributor=self.contributor)
+        collection2 = Collection.objects.create(name="Collection", contributor=self.contributor)
+        self.assertEqual(collection1.slug, 'collection')
+        self.assertEqual(collection2.slug, 'collection-2')
+
+        request = factory.patch(self.detail_url(collection1.pk), {'name': 'Collection', 'slug': None}, format='json')
+        force_authenticate(request, self.admin)
+        response = self.collection_detail_viewset(request, pk=collection1.pk)
+        self.assertEqual(response.data['slug'], 'collection')
+
+        collection3 = Collection.objects.create(name="Collection", contributor=self.contributor)
+        self.assertEqual(collection3.slug, 'collection-3')
+
+        request = factory.patch(self.detail_url(collection2.pk), {'name': 'Collection', 'slug': None}, format='json')
+        force_authenticate(request, self.admin)
+        response = self.collection_detail_viewset(request, pk=collection2.pk)
+        self.assertEqual(response.data['slug'], 'collection-4')
```

Comment:
This is probably not the desired behaviour and should be fixed at some point, so i think it is not a good idea to include it in tests.

Reply:
Is it the problem that the new slug should be `collection-2`?

Reply:
Fixed in https://github.com/genialis/resolwe/pull/527/commits/97a6c69854191f3024e4b8ae6aaa006a34d3ea16

Reply:
It is not consistent with the first use-case in this test, where slugs stays the same. We should at least unify the behaviour.